lib/ansible/executor/play_iterator.py

```diff
@@ -242,16 +240,10 @@ def get_host_state(self, host):
         return self._host_states[host.name].copy()
 
     def cache_block_tasks(self, block):
-        def _cache_portion(p):
-            for t in p:
-                if isinstance(t, Block):
-                    self.cache_block_tasks(t)
-                elif t._uuid not in self._task_uuid_cache:
-                    self._task_uuid_cache[t._uuid] = t
-
-        for portion in (block.block, block.rescue, block.always):
-            if portion is not None:
-                _cache_portion(portion)
+        # now a noop, we've changed the way we do caching and finding of
+        # original task entries, but just in case any 3rd party strategies
+        # are using this we're leaving it here for now
+        return
```

Comment:
Should we issue a deprecation here and in get_original_task?  (bcoca advised me to do that for something else that maintained a backwards-compat API while our code did something different.).

Reply:
@jimi-c is the intention that "nothing *should* be using this, but just in case someone out in the ether *is* using then we're not going to break them"? .... that's how I understood it but would like to clarify given @abadger's question.

Reply:
Yep that's correct.